code_challenges/code_wars/longest_collatz_sequence.py

```diff
@@ -0,0 +1,22 @@
+def find_collatz_sequence(numbers):
+    numbers_with_sequence = {}
+    for number in numbers:
+        numbers_with_sequence[number] = 0
+        updated_number = number
+        while updated_number > 1:
+            updated_number = collatz_number(updated_number)
+            numbers_with_sequence[number] += 1
+    return numbers_with_sequence
+
+
+def collatz_number(num):
+    if num % 2 == 0:
+        return num / 2
+    else:
+        return 3 * num + 1
+
+
+def find_longest_length(numbers):
+    return max(
+        find_collatz_sequence(numbers), key=lambda x: find_collatz_sequence(numbers)[x]
+    )
```

Comment:
You shouldn't call  `find_collatz_sequence` twice here.
```suggestion
    numbers_with_sequence =  find_collatz_sequence(numbers)
    return max(
        numbers_with_sequence, key=lambda x: numbers_with_sequence[x]
    )
```

Reply:
i had it that way and then thought "i shouldnt make a variable when i dont need to." But I see how my original thought to have the function call in a variable is better.